core/flyout_vertical.js

```diff
@@ -546,27 +546,45 @@ Blockly.VerticalFlyout.prototype.createCheckbox_ = function(block, cursorX,
  * @private
  */
 Blockly.VerticalFlyout.prototype.checkboxClicked_ = function(checkboxObj) {
+  var context = this;
   return function(e) {
-    var oldValue = checkboxObj.clicked;
-    var newValue = !oldValue;
-    checkboxObj.clicked = newValue;
-
-    if (checkboxObj.clicked) {
-      Blockly.utils.addClass((checkboxObj.svgRoot), 'checked');
-    } else {
-      Blockly.utils.removeClass((checkboxObj.svgRoot), 'checked');
-    }
-
-    Blockly.Events.fire(new Blockly.Events.Change(
-        checkboxObj.block, 'checkbox', null, oldValue, newValue));
-
+    context.setCheckboxState(checkboxObj.block.id, !checkboxObj.clicked);
```

Comment:
Please use the poor man's arrow function here, and `.bind(this)` on the function definition.